### DISCUSSION TITLE
Schema registry installation on bare metal

### DISCUSSION THREAD

**Discussion by Woj-Teh** [Apicurio/apicurio-registry#5601] [Q&A]
Hello!
I have kafka installed on bare metal. I would also like to install schema registry. Unfortunately, the documentation and many sources only describe installation on Docker or kuebrnetes cluster. Is there a way to install schema registry on bare metal?
Thanks for help!

**Comment by Ganesh-403** [comment]
@Woj-Teh Yes, you can absolutely run Apicurio Registry on bare metal. Docker and Kubernetes are the most common deployment options because they're convenient, but they are **not** a requirement.

Since Apicurio Registry 3.x is a Quarkus application, you can run it as a regular Java application using the packaged JAR (or native executable if you've built one). The runtime configuration is provided via `application.properties`, Java system properties, or environment variables, just as with any other Quarkus application. The official configuration reference documents these runtime options independently of Docker or Kubernetes. :contentReference[oaicite:0]{index=0}

A typical bare-metal deployment would look like:

1. Install a supported JDK (Java 17+ for the 3.x releases).
2. Obtain the Apicurio Registry distribution.
3. Configure your storage backend (PostgreSQL, SQL Server, KafkaSQL, etc.).
4. Configure authentication (optional).
5. Start the application as a normal Java process, for example:

```bash
java -jar apicurio-registry-app.jar
```

or manage it with a service manager such as `systemd`.

For example, if you're using PostgreSQL:

```properties
apicurio.storage.kind=sql
apicurio.storage.sql.kind=postgresql
apicurio.datasource.url=jdbc:postgresql://db-host:5432/registry
apicurio.datasource.username=registry
apicurio.datasource.password=secret
```

The Registry will then connect to your external database and create its required metadata tables on startup (assuming the configured user has the necessary permissions). :contentReference[oaicite:1]{index=1}

The Docker images are essentially a packaged way of running this same Quarkus application with environment variables already wired up—they aren't required by the Registry itself.

One question:

- Are you planning to install it directly on a Linux server (e.g. with `systemd`), or are you looking for a packaged distribution (RPM/DEB) for your operating system?

That will determine the most appropriate deployment approach.

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.
